### ISSUE TITLE
Axis error with PS3 controller

### ISSUE THREAD

**Issue by StarlightNL** [opentk/opentk#298]
Whenever I try to use a PS3 controller the program gives me a "Argument is out of range" error on the Axis. OpenTK supports up to 10 axis but the PS3 controller got 29 axis(Bluetooth and USB got different axis bindings)

```
System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: axis
  at OpenTK.Input.JoystickState.SetAxis (JoystickAxis axis, Int16 value) [0x00000] in <filename unknown>:0
  at OpenTK.Platform.Linux.LinuxJoystick.PollJoystick (OpenTK.Input.JoystickDevice`1 js) [0x00000] in <filename unknown>:0
  at OpenTK.Platform.Linux.LinuxJoystick.OpenTK.Input.IJoystickDriver2.GetState (Int32 index) [0x00000] in <filename unknown>:0
  at OpenTK.Input.Joystick.GetState (Int32 index) [0x00000] in <filename unknown>:0
  at OpenTK.Platform.MappedGamePadDriver.GetState (Int32 index) [0x00000] in <filename unknown>:0
  at OpenTK.Input.GamePad.GetState (Int32 index) [0x00000] in <filename unknown>:0
  at ControllerTest.MainClass.Main (System.String[] args) [0x00007] in /home/starlight/MonoProjects/ControllerTest/ControllerTest/Program.cs:15
```

Code at line 15 is 
`JoystickState joyStick = Joystick.GetState(0);`

I tried changing the JoystickAxis file to include more axis but then I get an other error

```
Unhandled Exception:
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
  at System.Collections.Generic.List`1+Enumerator[OpenTK.Platform.Linux.LinuxJoystickDetails].MoveNext () [0x00000] in <filename unknown>:0 
  at OpenTK.Platform.Linux.LinuxJoystick.Dispose (Boolean manual) [0x00000] in <filename unknown>:0 
  at OpenTK.Platform.Linux.LinuxJoystick.Finalize () [0x00000] in <filename unknown>:0 
```

**Comment by thefiddler** [comment]
There's something weird happening here. Why does the PS3 controller report 29 axes? That makes very little sense.

I'm guessing that it has analogue rather than digital buttons (i.e. its buttons report how strongly they are pressed). Does it make sense to support that functionality? I think SDL2 simply maps analogue buttons to digital on/off buttons.

It would be useful to see a debug log from OpenTK with the PS3 controller connected.

**Comment by StarlightNL** [comment]
I am not using a PS3 controller anymore on a linux machine but the reason why it is reporting 29 axis is because every key is also mapped as a analog button, but only in bluetooth mode. On Windows the problem didn't exist.

**Comment by varon** [comment]
Hopefully fixed by #453.
